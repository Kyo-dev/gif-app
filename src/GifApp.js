import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifCollection } from "./components/GifCollection";

const GifApp = ({defaultCategories = []}) => {
  const [categories, setCategories] = useState(defaultCategories);
  return (
    <>
      <h2>Gif App</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((c, i) => (
          <GifCollection key={i} category={c} />
        ))}
      </ol>
    </>
  );
};

export default GifApp;
