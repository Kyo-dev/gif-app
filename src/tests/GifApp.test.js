import'@testing-library/jest-dom'
import React from 'react'
import {shallow} from 'enzyme'
import GifApp from '../GifApp'

describe('test in GifApp', () => {

    
    test('should component OK', () => {
        const wrapper = shallow(<GifApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should list categories  ', () => {
        const categories = ['samurai X', 'Sakura card captor'];
        const wrapper = shallow(<GifApp defaultCategories = {categories} />);
        expect(wrapper.find('GifCollection').length).toBe(categories.length);

    })
    
})
