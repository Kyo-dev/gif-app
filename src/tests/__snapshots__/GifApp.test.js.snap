// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifApp should component OK 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
